wasp-app-runner/src/dependencies.ts

```diff
@@ -0,0 +1,38 @@
+import { exec } from "child_process";
+
+import { log } from "./logging.js";
+import { installWaspCli } from "./waspCli.js";
+
+export async function checkAndSetupDependencies({
+  buildWaspCli,
+}: {
+  buildWaspCli: boolean;
+}) {
+  const requiredCommands = ["docker"];
+  if (buildWaspCli) {
+    requiredCommands.push("cabal");
+  }
+
+  for (const cmd of requiredCommands) {
+    if (!(await commandExists(cmd))) {
+      log(
+        "setup",
+        "error",
+        `Required command '${cmd}' not found. Please install it.`
+      );
+      process.exit(1);
+    }
+  }
+
+  if (buildWaspCli) {
+    await installWaspCli();
```

Comment:
We talked in person and agreed we'll remove the installation details from this project (Unix philosophy and all that).